packages/graphql/tests/integration/directives/fulltext.int.test.ts

```diff
@@ -74,672 +95,2324 @@ describe("@fulltext directive", () => {
 
         await driver.close();
     });
-
-    test("should create index if it doesn't exist and then query using the index", async () => {
+    describe("Functionality", () => {
         // Skip if multi-db not supported
         if (!MULTIDB_SUPPORT) {
             console.log("MULTIDB_SUPPORT NOT AVAILABLE - SKIPPING");
             return;
         }
 
-        const title = generate({ readable: true, charset: "alphabetic" });
-        const indexName = generate({ readable: true, charset: "alphabetic" });
-        const type = generateUniqueType("Movie");
-
-        const typeDefs = gql`
-            type ${type.name} @fulltext(indexes: [{ name: "${indexName}", fields: ["title"] }]) {
-                title: String!
-            }
-        `;
-
-        const neoSchema = new Neo4jGraphQL({ typeDefs });
-        const schema = await neoSchema.getSchema();
-
-        await expect(
-            neoSchema.assertIndexesAndConstraints({
+        let session: Session;
+        let neoSchema: Neo4jGraphQL;
+        let generatedSchema: GraphQLSchema;
+        let personType: UniqueType;
+        let movieType: UniqueType;
+        let personTypeLowerFirst: string;
+        let queryType: string;
+
+        const person1 = {
+            name: "this is a name",
+            born: 1984,
+        };
+        const person2 = {
+            name: "This is a different name",
+            born: 1985,
+        };
+        const person3 = {
+            name: "Another name",
+            born: 1986,
+        };
+        const movie1 = {
+            title: "Some Title",
+            description: "some other description",
+            released: 2001,
+        };
+        const movie2 = {
+            title: "Another Title",
+            description: "this is a description",
+            released: 2002,
+        };
+
+        beforeEach(async () => {
+            personType = generateUniqueType("Person");
+            movieType = generateUniqueType("Movie");
+            queryType = `${personType.plural}Fulltext${upperFirst(personType.name)}Index`;
```

Comment:
The unique type class have an `operations` object to generate the top-level operations so we have an unified place to generate these instead of having this logic scattered around. Consider moving this to that class so we can have something like:
```suggestion
            queryType = personType.operations.fulltext;
```

Reply:
Unfortunately, this depends on the index name. These are defined separately and there can be several for each type/node so it's not really possible to do it like that